test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -87,10 +88,15 @@ private void assertCommandBehavior(String inputCommand,
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
 
+        // Get command type
+        Command cmd = new Parser().parseCommand(inputCommand);
+
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (cmd.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
```

Comment:
it would also be good to check if the command that is NOT mutating indeed doesn't save the file